hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.controller.js

```diff
@@ -362,32 +306,202 @@ define([
+     *
+     * @param {Number} index the index of the file in the section list.
+     * @param {String} entityName the section name where the file is stored.
+     */
+    function fileMoveToTrash (index, entityName) {
+      var entityFiles = $scope.files[entityName];
+      var entityFilesTrash = $scope.filesTrash[entityName];
+
+      entityFilesTrash.push(entityFiles[index]);
+      entityFiles.splice(index, 1);
+    }
+
+    /**
+     * Validates that all files are within the max file size limit.
+     */
+    function filesValidate () {
+      var entityName, fieldName, i, len, uploaderEntity, uploaderEntityField, uploaderEntityFieldQueue;
+      var fileMaxSize = $scope.fileMaxSize;
+      var uploader = $scope.uploader;
+      var isValid = true;
+
+      for (entityName in uploader) {
+        uploaderEntity = uploader[entityName];
+
+        for (fieldName in uploaderEntity) {
+          i = 0;
+          len = uploaderEntityFieldQueue.length;
+          uploaderEntityField = uploaderEntity[fieldName];
+          uploaderEntityFieldQueue = uploaderEntityField.queue;
+
+          for (; i < len && isValid; i++) {
+            isValid = uploaderEntityFieldQueue[i].file.size < fileMaxSize;
+          }
+        }
+      }
+
+      $scope.contractForm.$setValidity('maxFileSize', isValid);
+    }
+
+    /**
+     * Initializes files and uploader fields. For the files it creates a trash
+     * container in case the file is deleted and for uploaders it adds validators
+     * to run after file uploading.
+     */
+    function initFilesAndUploaders () {
+      angular.copy(files, $scope.files);
+      angular.forEach($scope.files, function (entityFiles, entityName) {
+        $scope.filesTrash[entityName] = [];
+      });
+      angular.forEach($scope.uploader, function (entity) {
+        angular.forEach(entity, function (field) {
+          field.onAfterAddingAll = function () {
+            $scope.filesValidate();
+          };
+        });
+      });
+    }
+
+    /**
+     * Determines if the given contract date has changed from its original value.
+     *
+     * @param {String} dateName either "period_start_date" or "period_end_date".
+     * @return {Boolean}
+     */
+    function hasContractDateChanged (dateName) {
+      var originalDate = entity.details[dateName];
+      var currentDate = $scope.entity.details[dateName];
+      var originalDateIsEmptyAndCurrentIsNot = _.isEmpty(originalDate) && !_.isEmpty(currentDate);
+      var currentDateIsDifferentFromOriginal = moment.isDate(currentDate) &&
+        !moment.utc(originalDate).isSame(moment.utc(currentDate), 'day');
+
+      return originalDateIsEmptyAndCurrentIsNot || currentDateIsDifferentFromOriginal;
+    }
+
+    /**
+     * Initializes the form entity fields by having a copy of the form entity
+     * and converting date strings into date objects.
+     */
+    function initFormEntity () {
+      angular.copy(entity, $scope.entity);
```

Comment:
In that case I would add `var details = $scope.entity.details;` at the top, but this is up to you completely.

Reply:
The "entity" object has many properties, if I store each one in a different variable I'd need to change all the references. I would avoid that if I could :(